File: src/api/BoardingTracker.Domain/Entities/Candidate.cs
namespace BoardingTracker.Domain.Entities
{
    public partial class Candidate
    {
        public Candidate()
        {
            Interviews = new HashSet<Interview>();
        }

        public Guid Id { get; set; }
        public string FirstName { get; set; } = null!;
        public string LastName { get; set; } = null!;
        public string PhoneNumber { get; set; } = null!;
        public string? Biography { get; set; }
        public string ResumeUrl { get; set; } = null!;
        public Guid? UserId { get; set; }

        public virtual User? User { get; set; }
        public virtual ICollection<Interview> Interviews { get; set; }
    }
}

File: src/frontend/src/app/candidates/models/candidate.ts
export class Candidate {
    public firstName: string | null = null;
    public lastName: string | null = null;
    public phoneNumber: string | null = null;
    public profileImageUrl: string | null = null;
    public biography: string | null = null;
    public resumeUrl: string | null = null;
    public email: string | null = null;
    userId: string | null = null;

    constructor(
        firstName: string,
        lastName: string,
        phoneNumber: string,
        biography: string,
        profileImageUrl: string,
        resumeUrl: string,
        email: string,
        userId: string) {
    this.firstName = firstName;
    this.lastName = lastName;
    this.phoneNumber = phoneNumber;
    this.biography = biography;
    this.profileImageUrl = profileImageUrl;
    this.resumeUrl = resumeUrl;
    this.email = email;
    this.userId = userId;
}

}